### PR TITLE
Payments

### PR DESCRIPTION
**Pull request by apofema**

### CODE REVIEW

File: Sources/Run/Routes/PostbackHandler.swift [hunk 1]
```diff
@@ -155,15 +155,21 @@ extension Droplet {
...
         let botHostName = getBotHostName(config)
         
-        let url = "\(botHostName)/web?host=\(host)&user_id=\(subscriber.fb_messenger_id)&price=\(price)&product=\(product)"
+        let url = "\(botHostName)/web?host=\(host)&user_id=\(subscriber.fb_messenger_id)&price=\(price)&product=\(product)&event=1"
+        let url2 = "\(botHostName)/web?host=\(host)&user_id=\(subscriber.fb_messenger_id)&price=\(price)&product=\(product)&event=2"
         
-        let buttonClaimSpot = ["type": "web_url", "url": url, "messenger_extensions": "true", "title": "Claim your spot now"]
-        let pollResults = drop.carouselElement(title: "Join Hanna Lee show",
+        let buttonClaimSpot = ["type": "web_url", "url": url, "messenger_extensions": "true", "title": "Book the class"]
+        let buttonClaimSpot2 = ["type": "web_url", "url": url2, "messenger_extensions": "true", "title": "Book the class"]
```
Comment (bsuvorov) [inline Sources/Run/Routes/PostbackHandler.swift]: wow, we offer two classes now? would be nice to have a screenshot or link to the task. i couldn't find task in trello that describes this code.

File: Sources/Run/Routes/PostbackHandler.swift [hunk 2]
```diff
@@ -155,15 +155,21 @@ extension Droplet {
...
+        let pollResults = drop.carouselElement(title: "Friday May 17th 7pm",
```
Comment (bsuvorov) [inline Sources/Run/Routes/PostbackHandler.swift]: p2: why pollResults? Do those link take them to booking or polling?

If booking, let's call it bookCardOne,  bookCardTwo

File: Sources/Run/Services/ProductPaymentHandler.swift [hunk 3]
```diff
@@ -98,17 +99,19 @@ extension Droplet {
             return Response(status: .noContent, body: "")
         }
         
-        guard result.status == .ok else {
+        if result.status != .ok {
```
Comment (bsuvorov) [inline Sources/Run/Services/ProductPaymentHandler.swift]: p2: why change  what's not broken? do you not like guard stylistically?

File: Sources/Run/Services/ProductPaymentHandler.swift [hunk 4]
```diff
@@ -98,17 +99,19 @@ extension Droplet {
...
             return Response(status: result.status, body: message.makeBody())
         }
-
-        self.send(message: "EAH!!! You’re on the next show! We will reach out with a few potential times", senderId: subscriber.fb_messenger_id, messagingType: .NON_PROMOTIONAL_SUBSCRIPTION)
-        
-        return result
+        else
+        {
+            analytics?.logAnalytics(event: .CompletedToPurchaseTheShow, for: subscriber, eventValue: event.string)
+            self.send(message: "EAH!!! You’re on the next show! We will reach out with a few potential times", senderId: subscriber.fb_messenger_id, messagingType: .NON_PROMOTIONAL_SUBSCRIPTION)
```
Comment (bsuvorov) [inline Sources/Run/Services/ProductPaymentHandler.swift]: `EAH!!!` - must be nadav's copy? Maybe `YEAH!!!` ?

File: Sources/Run/Routes/PostbackHandler.swift (hunk 1 above)
Comment (apofema) [reply Sources/Run/Routes/PostbackHandler.swift]: sent you to slack

File: Sources/Run/Routes/PostbackHandler.swift (hunk 2 above)
Comment (apofema) [reply Sources/Run/Routes/PostbackHandler.swift]: Yep, I forgot to change it. Will fix it

File: Sources/Run/Services/ProductPaymentHandler.swift (hunk 3 above)
Comment (apofema) [reply Sources/Run/Services/ProductPaymentHandler.swift]: so, what happened recently, when I opened the page, and did nothing for a few minutes I've received the "EAH! ..." It seems like the time out send the ok status as well. So I fixed that and was about to test but have got a merge conflict :) Which I'm trying to resolve on staging now.

**Pull request merged**